src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -824,7 +833,7 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
             bool hasInitializer = (_propertyFlags & Flags.HasInitializer) != 0;
             if (hasInitializer)
             {
-                CheckInitializer(IsAutoProperty, ContainingType.IsInterface, IsStatic, Location, diagnostics);
+                CheckInitializer(HasNonErrorBackingField, ContainingType.IsInterface, IsStatic, Location, diagnostics);
```

Comment:
By `NonError` I didn't mean "any error". The error I mean is only that we have a backing field for an old-style regular property that has an initializer.

Reply:
> By NonError I didn't mean "any error". The error I mean is only that we have a backing field for an old-style regular property that has an initializer.

Naming should clear to avoid leaving too much space for interpretation. Or comments should be added to explain.

Reply:
@AlekseyTs I think renaming it to `IsAutoOrSemiAutoProperty` will make sense for all call-sites. But we wanted to avoid the term `Auto` in implementation as much as possible. Is using `IsAutoOrSemiAutoProperty` name fine? If not, can we find a good name such that we keep it one property for all the current callsites?

Reply:
> especially that it affects many somewhat unrelated places in implementation

All callers want to check if we have a backing field, so they feel related to me. It was probably just the name that's not clear enough. Do you agree?

Reply:
> All callers want to check if we have a backing field, so they feel related to me. It was probably just the name that's not clear enough. Do you agree?

It feels like the logic might not be the same for all call sites.